cantools/db/formats/dbc.py

```diff
@@ -374,17 +374,19 @@ def _dump_messages(database):
     for message in database.messages:
         msg = []
         fmt = 'BO_ {frame_id} {name}: {length} {senders}'
-        msg.append(fmt.format(frame_id=message.frame_id,
+        msg.append(fmt.format(frame_id=message.frame_id | (1 << 31 if message.is_extended_frame else 0),
                               name=message.name,
                               length=message.length,
                               senders=' '.join(message.senders)))
 
         for signal in message.signals[::-1]:
-            fmt = (' SG_ {name} : {start}|{length}@{byte_order}{sign}'
+            fmt = (' SG_ {name}{muxType}{muxId} : {start}|{length}@{byte_order}{sign}'
                    ' ({scale},{offset})'
                    ' [{minimum}|{maximum}] "{unit}" {receivers}')
             msg.append(fmt.format(
                 name=signal.name,
+                muxType=' M' if signal.is_multiplexer else ' m' if signal.multiplexer_ids != None else '',
+                muxId='' if signal.multiplexer_ids == None else signal.multiplexer_ids[0],
```

Comment:
Lines too long. I think I prefer creating a local function `get_mux()` or similar that returns the mux string.